tests/integration/tests/no_istiod_test.go

```diff
@@ -1,24 +1,37 @@
 package tests
 
 import (
+	"context"
 	"os/exec"
 	"strconv"
 	"strings"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
 
 	"github.com/kiali/kiali/log"
 	"github.com/kiali/kiali/tests/integration/utils"
 )
 
+const kialiNamespace = "istio-system"
+
 var ocCommand = utils.NewExecCommand()
 
-func update_istio_api_enabled(value bool) {
+func update_istio_api_enabled(value bool, kubeClientSet kubernetes.Interface, ctx context.Context) {
 	original := !value
 
-	cmdGetProp := ocCommand + " get cm kiali -n istio-system -o yaml | grep 'istio_api_enabled'"
+	// Restart kiali pod
+	// Get kiali pod name
+	cmdGetPodName := ocCommand + " get pods -o name -n " + kialiNamespace + " | egrep kiali | sed 's|pod/||'"
+	kialiPodName, err2 := exec.Command("bash", "-c", cmdGetPodName).Output()
+	podName := strings.Replace(string(kialiPodName), "\n", "", -1)
+	log.Debugf("Kiali pod name: %s", podName)
```

Comment:
Why are we using shell commands to get/update kube objects? This wasn't part of your changes, I'm just curious.

Reply:
Well, nothing specific, as we are not testing the kube client and the shell is also available, is other possibility.

Reply:
But, is there is any issue I can change it.

Reply:
It'll be harder to check error output with running shell commands and harder to reuse code but save that for another change.